### PR TITLE
inform6: init at 6.34-6.12.2

### PR DESCRIPTION
**Pull request by ddelabru**
###### Motivation for this change

nixpkgs already has Inform 7, which is a lovely design system for creating interactive fiction (text adventures) using a natural language compiler and a rich IDE. However, underlying Inform 7 is a C-like language, Inform 6, which has a much simpler toolchain and its own compiler and standard libraries, which I have packaged for nix here.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ddelabru** [comment]
@DavidGriffith, I welcome your thoughts as upstream maintainer for Inform 6.

@mbx6spp, also interested if you have any feedback as the maintainer of the nix package for Inform 7.

**Comment by ddelabru** [comment]
@ofborg build

**Comment by ddelabru** [comment]
@ofborg build inform6

File: pkgs/development/compilers/inform6/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  src = fetchurl  {
+    url = "https://ifarchive.org/if-archive/infocom/compilers/inform6/source/inform-${version}.tar.gz";
+    sha256 = "c149f143f2c29a4cb071e578afef8097647cc9e823f7fcfab518ac321d9d259f";
+  };
+
+  buildInputs = [ perl ];
+
+  buildPhase = ''
+    make PREFIX=$out
+  '';
```
Comment (Mic92) [inline pkgs/development/compilers/inform6/default.nix:16]:
```suggestion
  buildFlags = [ "PREFIX=${placeholder "out"}" ];
```

File: pkgs/development/compilers/inform6/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  installPhase = ''
+    make install PREFIX=$out
+  '';
```
Comment (Mic92) [inline pkgs/development/compilers/inform6/default.nix:20]: I think this also applies to install phase? If not use `installFlags` just like `buildFlags`

```suggestion
```

File: pkgs/development/compilers/inform6/default.nix (hunk 1 above)
Comment (ddelabru) [reply pkgs/development/compilers/inform6/default.nix]: Thanks for the tip; I amended my commit to use `buildFlags`

File: pkgs/development/compilers/inform6/default.nix (hunk 2 above)
Comment (ddelabru) [reply pkgs/development/compilers/inform6/default.nix]: I tested this and it turns out I do need both `buildFlags` and `installFlags`. This is reflected in the latest version of my changes.

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -4370,6 +4370,8 @@ in
 
   inetutils = callPackage ../tools/networking/inetutils { };
 
+  inform6 = callPackage ../development/compilers/inform6 { };
+
   inform7 = callPackage ../development/compilers/inform7 { };
```
Comment (Mic92) [inline pkgs/top-level/all-packages.nix:4375]: What is the difference between the two?

**Comment by Mic92** [comment]
Are both projects still maintained?

**Comment by ddelabru** [comment]
@Mic92, yes, both projects are still maintained. Inform 6 is maintained by David Griffith, who doesn't appear to be active on GitHub and has moved all activity on Inform 6, frotz, and other projects over to gitlab.com.

**Comment by DavidGriffith** [comment]
This looks okay to me.

**Pull request merged**